### PR TITLE
Update dependencies and Hackage index-state

### PR DESCRIPTION
**Pull request by erikd**
Updated Hackage index-state is needed to fix https://github.com/input-output-hk/cardano-node/issues/1139

### CODE REVIEW

**Comment by rvl** [comment]
The failing tests are here: https://hydra.iohk.io/job/Cardano/ouroboros-network-pr-2229/required#tabs-constituents

1. `ouroboros-consensus-shelley-test` fails on linux/macos/windows.
2. `ouroboros-network-framework-tests` fails on macos only

**Comment by mrBliss** [comment]
> ~LGTM~ - it seems that some consensus tests are failing. @mrBliss

>     1. `ouroboros-consensus-shelley-test` fails on linux/macos/windows.

The golden tests for Shelley are failing because of changes in `cardano-ledger-specs`. @erikd Do you want me to fix them?

**Comment by erikd** [comment]
@mrBliss yes please.

**Comment by mrBliss** [comment]
> @mrBliss yes please.

On it. I'll first have to compile half the world, though :grin:.

**Comment by mrBliss** [comment]
Oh, https://github.com/input-output-hk/ouroboros-network/pull/2272 is already updating the dependency on `cardano-ledger-specs`. Let's merge that one before this one to avoid duplicating the work.

File: nix/sources.json [hunk 1]
```diff
@@ -17,10 +17,10 @@
         "homepage": null,
         "owner": "input-output-hk",
         "repo": "iohk-nix",
-        "rev": "a3eae81cbae000bbfdc2bc1880528512747a3a9c",
-        "sha256": "1jairn2vp43z36j0b1fi2nmnfpadff217dqv5bjjxm5v099g0gsw",
+        "rev": "02569d546f72216d63e406f80dd469bfc8381946",
```
Comment (nc6) [inline nix/sources.json:20]: This is actually a reversion

File: nix/sources.json [hunk 2]
```diff
@@ -5,10 +5,10 @@
         "homepage": "https://input-output-hk.github.io/haskell.nix",
         "owner": "input-output-hk",
         "repo": "haskell.nix",
-        "rev": "59cf05606e7efbbc4741ae28fd8cc610cec94eb8",
-        "sha256": "1dpk49z97ld6a9ggy9ph6apk1ccgjwwnc4pzirh6galbcci4rqap",
+        "rev": "edc948e0e8dc9b71b22b61ac06be2b4d5f54591d",
```
Comment (nc6) [inline nix/sources.json:8]: This is also a reversion

**Comment by nc6** [comment]
bors merge

**Pull request merged**